Classes/Template.php

```diff
@@ -158,9 +158,12 @@ public function createOrFetchAndApply(NodeInterface $parentNode, array $context)
                 $flowQuery = new FlowQuery(array($parentNode));
                 $node = $flowQuery->find($name)->get(0);
             }
+            $type = $this->type;
```

Comment:
Could we move the new type evaluation block inside the if statement right before the $this->nodeOperations->create, as it's only needed when a new node has to be created?

Reply:
yes 👍 ;)